### PR TITLE
Add createAndExerciseCmd to triggers

### PR DESCRIPTION
**Pull request by shayne-fletcher**
Fixes https://github.com/digital-asset/daml/issues/5956.

### CODE REVIEW

File: triggers/daml/Daml/Trigger.daml [hunk 1]
```diff
@@ -110,6 +112,22 @@ dedupCreate t = do
   unless (any ((Some t ==) . fromCreate) cmds) $
     void $ emitCommands [createCmd t] []
 
+-- | Create the template and exercise a choice on it it’s not already in the list of commands
+-- in flight (it will still be created if it is in the ACS).
+--
+-- Note that this will send the create as a single-command transaction.
+-- If you need to send multiple commands in one transaction, use
+-- `emitCommands` with `createCmd` and handle filtering yourself.
+dedupCreateAndExercise : (Eq t, Eq c, Template t, Choice t c r) => t -> c -> TriggerA ()
```
Comment (cocreature) [inline triggers/daml/Daml/Trigger.daml:122]: fwiw I’m considering removing the `dedupX` functions completely in favor of the pending commands in `emitCommands`. While they seem convenient on simple examples, I think they also make it easy to get the condition you want to deduplicate on wrong and they don’t scale to multi-command transactions.

Let’s add it for now so we are consistent. If you have an opinion on this, I’d be interested in hearing it (probably not on this PR though, it’s completely unrelated :))

File: triggers/tests/daml/CreateAndExercise.daml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+module CreateAndExercise where
+
+import DA.Next.Map (Map)
+import Daml.Trigger
+
+createAndExerciseTrigger : Trigger Int
+createAndExerciseTrigger = Trigger
+  { initialize = \_acs -> 3
+  , updateState = \_acs msg allowedFail -> case msg of
```
Comment (cocreature) [inline triggers/tests/daml/CreateAndExercise.daml:13]: The retries here look like you’ve copied them from another example but aren’t actually using them. Can we simplify the test case?

Comment (shayne-fletcher) [reply triggers/tests/daml/CreateAndExercise.daml:13]: I did and no I'm not. Indeed, I'll have a look.

Comment (cocreature) [reply triggers/tests/daml/CreateAndExercise.daml:13]: You can probably just change the trigger to `Trigger ()` and turn this into `\_ _ _ -> ()`

Comment (shayne-fletcher) [reply triggers/tests/daml/CreateAndExercise.daml:13]: Done.

**Pull request merged**